controllers/front/DiscountController.php

```diff
@@ -39,15 +39,26 @@ public function initContent()
     {
         parent::initContent();
 
-        $cart_rules = CartRule::getCustomerCartRules($this->context->language->id, $this->context->customer->id, true, false);
+        $cart_rules = CartRule::getCustomerCartRules($this->context->language->id, $this->context->customer->id, true, true, true);
```

Comment:
just upgraded a shop to 1.6.1.6 and noticed **all Vouchers can be viewed by the customers** :(

#5797 thx! helped though.